exporters/otlp/otlp.go

```diff
@@ -102,103 +97,40 @@ var (
 	errContextCanceled = errors.New("context canceled")
 )
 
-// Start dials to the collector, establishing a connection to it. It also
-// initiates the Config and Trace services by sending over the initial
+// Start dials to the collector for both traces and metrics, establishing connections.
+// It also initiates the Config and Trace services by sending over the initial
 // messages that consist of the node identifier. Start invokes a background
-// connector that will reattempt connections to the collector periodically
+// connector that will reattempt connections to the traces/metrics collector periodically
 // if the connection dies.
+// The connection is managed through otlpConnection.
 func (e *Exporter) Start() error {
 	var err = errAlreadyStarted
 	e.startOnce.Do(func() {
 		e.mu.Lock()
 		e.started = true
-		e.disconnectedCh = make(chan bool, 1)
 		e.stopCh = make(chan bool)
-		e.backgroundConnectionDoneCh = make(chan bool)
 		e.mu.Unlock()
 
-		// An optimistic first connection attempt to ensure that
-		// applications under heavy load can immediately process
-		// data. See https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/pull/63
-		if err := e.connect(); err == nil {
-			e.setStateConnected()
-		} else {
-			e.setStateDisconnected(err)
-		}
-		go e.indefiniteBackgroundConnection()
-
-		err = nil
+		err = e.startExporterConnections(e.stopCh)
 	})
 
 	return err
 }
 
-func (e *Exporter) prepareCollectorAddress() string {
-	if e.c.collectorAddr != "" {
-		return e.c.collectorAddr
-	}
-	return fmt.Sprintf("%s:%d", DefaultCollectorHost, DefaultCollectorPort)
-}
-
-func (e *Exporter) enableConnections(cc *grpc.ClientConn) error {
+func (e *Exporter) startExporterConnections(stopCh chan bool) error {
 	e.mu.RLock()
 	started := e.started
 	e.mu.RUnlock()
-
 	if !started {
 		return errNotStarted
 	}
 
-	e.mu.Lock()
-	// If previous clientConn is same as the current then just return.
-	// This doesn't happen right now as this func is only called with new ClientConn.
-	// It is more about future-proofing.
-	if e.grpcClientConn == cc {
-		e.mu.Unlock()
-		return nil
-	}
-	// If the previous clientConn was non-nil, close it
-	if e.grpcClientConn != nil {
-		_ = e.grpcClientConn.Close()
-	}
-	e.grpcClientConn = cc
-	e.traceExporter = coltracepb.NewTraceServiceClient(cc)
-	e.metricExporter = colmetricpb.NewMetricsServiceClient(cc)
-	e.mu.Unlock()
-
+	// TODO @sprisca: would a signal on the stop chanel stop both connections?
```

Comment:
I am not sure what the stopCh is being used for. I have not seen any actual write to this channel, which would stop the background connections (maybe I'm missing something?)
Either way, now that there are two connections there should be two channels also, since one connection will miss the stop signal if the other one consumes it first. What do you think?